std/datetime/date.d

```diff
@@ -6594,12 +6594,12 @@ public:
             $(REF DateTimeException,std,datetime,date) if the given day is an
             invalid day of the year.
       +/
-    @property void dayOfYear(int day) @safe pure
+    @property void dayOfYear(int day) @safe pure nothrow @nogc
     {
         immutable int[] lastDay = isLeapYear ? lastDayLeap : lastDayNonLeap;
 
         if (day <= 0 || day > (isLeapYear ? daysInLeapYear : daysInYear))
-            throw new DateTimeException("Invalid day of the year.");
+            assert(0, "Invalid day of the year.");
```

Comment:
I'm very well aware of https://github.com/dlang/druntime/pull/1710. As a matter of fact I intend to resurrect it, but haven't had much free time lately. However, I strongly believe that throwing is not appropriate in this situation.
Breaking changes are a spectrum and in this case no one's code will cease to compile. On the contrary, there are two immediate benefits:
* More of `std.datetime.date` become usable in `@nogc nothrow` code -> a game enabler
* People who have used anti-patterns such as `catch (Throwable)` or `catch (Exception)` would finally be able to fix lurking problem in their code.

@andralex, please advice.